### PR TITLE
Add Functionality For Authentication of Subscription Requests

### LINKED ISSUE #20: Functionality For Authentication of Subscription Request
Yea I was noticing that in Coinbase’s docs it says that you can authenticate a [subscription request](https://docs.pro.coinbase.com/?r=1#subscribe) in their web socket api to get additional attribution in channel messaging (from the link you have to scroll down a bit) but it didn’t seem like you could do that explicitly via your library.

### PR DESCRIPTION
**Pull request by theluckysuccess**
- Add Websocket subscription authentication
- Update demo in CoinbaseCli 
- Update Readme

Resolves #20

### CODE REVIEW

**Comment by zugaldia** [comment]
Thank you @theluckysuccess for the contribution! I'll review it this week so that we can make a new release over the weekend at the latest.

File: cli/src/main/kotlin/com/westwinglabs/coinbase/cli/CoinbaseCli.kt [hunk 1]
```diff
@@ -15,7 +15,7 @@ import org.apache.commons.cli.CommandLine
 import org.apache.logging.log4j.LogManager
 import java.io.File
 
-class CoinbaseCli : FeedListener() {
+open class CoinbaseCli : FeedListener() {
```
Comment (zugaldia) [inline cli/src/main/kotlin/com/westwinglabs/coinbase/cli/CoinbaseCli.kt:18]: Curious, why did you need to change the class to be `open`?

**Review by zugaldia — APPROVED** [review]
Looking good - only a small comment. We can merge as is and iterate, I don't want to block you from being able to use authentication.

File: cli/src/main/kotlin/com/westwinglabs/coinbase/cli/CoinbaseCli.kt (hunk 1 above)
Comment (Luckvery) [reply cli/src/main/kotlin/com/westwinglabs/coinbase/cli/CoinbaseCli.kt:18]: 1. I wanted to leverage all the methods that you created in `CoinbaseCli` without duplicating it
2. ...when all I really wanted to change was to overload the `onOpen` method in `CoinbaseCli` to take advantage of `SubscribeRequestAuthenticated`
3. ....but In Kotlin objects are closed by default. Essentially equivalent to objects created with `Final` in Java
4. So adding `open` to `CoinbaseCli` allowed me to create an anonymous object that inherited from it, which I could then overload its `onOpen` method [here](https://github.com/westwinglabs/coinbase-kotlin/pull/21/files#diff-f3b556d87f29468c42ead6aa82afa021ed65fed8e3faf85d1e0e3566ef8b069fR92).

Comment (theluckysuccess) [reply cli/src/main/kotlin/com/westwinglabs/coinbase/cli/CoinbaseCli.kt:18]: 1. I wanted to leverage all the methods that you created in `CoinbaseCli` without duplicating it
2. ...when all I really wanted to change was to overload the `onOpen` method in `CoinbaseCli` to take advantage of `SubscribeRequestAuthenticated`
3. ....but In Kotlin objects are closed by default. Essentially equivalent to objects created with `Final` in Java
4. So adding `open` to `CoinbaseCli` allowed me to create an anonymous object that inherited from it, which I could then overload its `onOpen` method [here](https://github.com/westwinglabs/coinbase-kotlin/pull/21/files#diff-f3b556d87f29468c42ead6aa82afa021ed65fed8e3faf85d1e0e3566ef8b069fR92).

**Comment by zugaldia** [comment]
Merged! Congrats @theluckysuccess on being the first external contributor to this project!
